coral/src/app/components/Form.tsx

```diff
@@ -448,12 +448,11 @@ NativeSelect.Skeleton = BaseNativeSelect.Skeleton;
 
 type ButtonProps = React.ComponentProps<typeof Button.Primary>;
 function _SubmitButton<T extends FieldValues>({
-  formContext: {
-    formState: { isSubmitting },
-  },
+  formContext: { formState },
   ...props
 }: ButtonProps & FormRegisterProps<T>) {
-  return <Button.Primary {...props} loading={isSubmitting} type="submit" />;
+  const loadingProp = props.loading || formState.isSubmitting;
```

Comment:
I think destructuring `isSubmitting` guaranteed we didn't find ourselves not subscribing to the changes in `formState`, as is described in the documentation regarding the `formState` `Proxy`: https://react-hook-form.com/docs/useform/formstate

> formState is wrapped with a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to improve render performance and skip extra logic if specific state is not subscribed to. Therefore make sure you invoke or read it before a render in order to enable the state update.

I *think* that doing this is also reading the value before render, but I'm only 99% sure haha! Destructuring in props is 100%.

```tsx
  const loadingProp = props.loading || formState.isSubmitting;
```

Reply:
Hm, I logged the value of `isSubmitting` (and `loadingProp`) while testing and it did update correctly (I thought because we're accessing it with `formState.isSubmitting`, but that's conditional. Will  destructure it, there's no real reason not to :D and better safe then sorry! (also, I did not test `isSubmitting` manually in the app).

Reply:
Updated: https://github.com/Aiven-Open/klaw/pull/1743/commits/104d2c2e28a54b4c200d0518cbbbcd66b2b72e06

Also checked again if the prop is set correctly (it is):
<img width="228" alt="Screenshot 2023-09-07 at 14 03 59" src="https://github.com/Aiven-Open/klaw/assets/943800/0de03a1f-9b5f-47a5-9b24-a3d2400ea0da">

Reply:
Yes, this is exteremely conservative and defensive programming haha! But better safe than sorry, desynced `formState` values seems like a very annoying bug to run into.